mock-plugin/src/test/java/io/mosip/signup/plugin/mock/service/MockProfileRegistryPluginImplTest.java

```diff
@@ -0,0 +1,272 @@
+        Map<String, Object> verifiedData = new HashMap<>();
+        verifiedData.put("individualId","1234567890");
+        JsonNode mockIdentity = objectMapper.valueToTree(verifiedData);
+        ProfileDto profileDto = new ProfileDto();
+        profileDto.setIndividualId("1234567890");
+        profileDto.setIdentity(mockIdentity);
+
+        MockIdentityResponse mockIdentityResponse=new MockIdentityResponse();
+        mockIdentityResponse.setStatus("CREATED");
+        ResponseWrapper<MockIdentityResponse> responseWrapper = new ResponseWrapper<>();
+        responseWrapper.setResponse(mockIdentityResponse);
+        ResponseEntity<ResponseWrapper<MockIdentityResponse>> responseEntity=new ResponseEntity<>(responseWrapper, HttpStatus.OK);
+
+        Mockito.when(restTemplate.exchange(
+                Mockito.anyString(),
+                Mockito.any(HttpMethod.class),
+                Mockito.any(),
+                Mockito.eq(new ParameterizedTypeReference<ResponseWrapper<MockIdentityResponse>>() {
+                }))).thenReturn(responseEntity);
+
+        ProfileResult result = mockProfileRegistryPlugin.createProfile("requestId", profileDto);
+        Assert.assertNotNull(result);
+        Assert.assertEquals(result.getStatus(), "CREATED");
+    }
+
+
+    @Test
+    public void createProfile_withInValidRequestAndProfileDto_thenFail() throws ProfileException {
+        // Arrange
+        ReflectionTestUtils.setField(mockProfileRegistryPlugin, "usernameField","individualId");
+        Map<String,String> verifiedData=new HashMap<>();
+        verifiedData.put("individualId","1234567890");
+        JsonNode mockIdentity = objectMapper.valueToTree(verifiedData);
+        ProfileDto profileDto = new ProfileDto();
+        profileDto.setIndividualId("123456789");
+        profileDto.setIdentity(mockIdentity);
+        try{
+            mockProfileRegistryPlugin.createProfile("requestId", profileDto);
+        }catch (ProfileException e){
+            Assert.assertEquals(ErrorConstants.IDENTIFIER_MISMATCH,e.getMessage());
+        }
+    }
+
+
+    @Test
+    public void getProfileCreateUpdateStatus_withValidRequestId_thenPass() throws ProfileException {
+        String requestId = "requestId123";
+
+        ProfileCreateUpdateStatus status = mockProfileRegistryPlugin.getProfileCreateUpdateStatus(requestId);
+
+        Assert.assertEquals(status,ProfileCreateUpdateStatus.COMPLETED);
+    }
+
+    @Test
+    public void getProfile_withValidIndividualId_thenPass() throws ProfileException {
+        String individualId = "1234567890";
+        ReflectionTestUtils.setField(mockProfileRegistryPlugin, "getIdentityEndpoint","http://localhost:8080/");
+
+        Map<String, Object> verifiedData = new HashMap<>();
+        verifiedData.put("email","123@email.com");
+        verifiedData.put("password","123456");
+        verifiedData.put("UIN","1234567890");
+        verifiedData.put("individualId",individualId);
```

Comment:
Kindly name the variable correctly. Why is this named as "verifiedData" ? its just identityData.

Reply:
renamed